main/src/main/scala/sbt/internal/server/BuildServerProtocol.scala

```diff
@@ -587,7 +588,12 @@ object BuildServerProtocol {
       workingDirectory = Some(baseDirectory.value),
       runJVMOptions = mainClass.jvmOptions,
       connectInput = connectInput.value,
-      envVars = envVars.value
+      envVars = envVars.value ++ mainClass.environmentVariables
```

Comment:
Here we concatenate the envVars of sbt process and ones submitted by client. So, my guess is that we don't need to put the envVars of sbt process in the `ScalaMainClass` data structure.